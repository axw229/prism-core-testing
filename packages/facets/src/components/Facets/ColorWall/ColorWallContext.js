// @flow
import React from 'react'
import noop from 'lodash/noop'
import type { Color } from 'src/shared/types/Colors.js.flow'
import { type GridBounds } from './ColorWall.flow'

export type ColorWallA11yContextProps = {
  a11yFocusCell: number[] | void,
  a11yFocusChunk: GridBounds,
  a11yFromKeyboard: boolean,
  a11yFocusOutline: boolean,
  a11yMaintainFocus: boolean
}

export const colorWallA11yContextDefault: ColorWallA11yContextProps = {
  a11yFocusCell: undefined,
  a11yFocusChunk: undefined,
  a11yFromKeyboard: false,
  a11yFocusOutline: false,
  a11yMaintainFocus: false
}

export type ColorWallContextProps = {
  activeColorRouteBuilderRef?: { current: (Color) => void },
  addButtonText?: string,
  autoHeight?: boolean,
  chunkClickable?: boolean,
  chunkMiniMap?: boolean,
  colorDetailPageRoot?: (Color) => string | string,
  colorWallBgColor: string,
  colorWallPageRoot?: (string | Color) => string | string,
  displayAddButton?: boolean,
  displayAddButtonText?: boolean,
  displayDetailsLink?: boolean,
  displayInfoButton?: boolean,
  hiddenSections?: string[],
  inactiveColorRouteBuilderRef?: { current: (Color) => void },
  isChipLocator?: boolean,
  loading: boolean,
  leftHandDisplay: boolean,
  routeType: string,
  swatchMaxSize: number,
  swatchMinSize: number,
  swatchMinSizeZoomed: number,
  swatchSizeZoomed: number,
  updateA11y: Function
}

export const colorWallContextDefault: ColorWallContextProps = {
  activeColorRouteBuilderRef: undefined,
  addButtonText: undefined,
  autoHeight: false,
  chunkClickable: false,
  chunkMiniMap: false,
  colorDetailPageRoot: undefined,
  colorWallBgColor: '#EEEEEE',
  colorWallPageRoot: undefined,
  displayAddButton: false,
  displayAddButtonText: false,
  displayDetailsLink: true,
  displayInfoButton: false,
  hiddenSections: [],
  inactiveColorRouteBuilderRef: undefined,
  isChipLocator: false,
  loading: false,
  leftHandDisplay: false,
  routeType: 'hash',
  swatchMaxSize: 33,
  swatchMinSize: 14,
  swatchMinSizeZoomed: 50,
  swatchSizeZoomed: 50,
  updateA11y: noop,
  viewColorText: false
}

const ColorWallContext = React.createContext<Object>({
  ...colorWallA11yContextDefault,
  ...colorWallContextDefault
})

export default ColorWallContext
