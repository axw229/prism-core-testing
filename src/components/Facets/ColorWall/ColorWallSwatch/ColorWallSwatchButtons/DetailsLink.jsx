// @flow
import React from 'react'
import { Link } from 'react-router-dom'
import { FormattedMessage } from 'react-intl'
import kebabCase from 'lodash/kebabCase'
import { type Color } from '../../../../../../src/shared/types/Colors'

type Props = {
  config: Object,
  detailsLink?: string,
  color: Color
}

const DetailsLink = (props: Props) => {
  const { config, detailsLink, color, ...other } = props

  // TODO: Future scope, the below shouldn't be driven by a data attribute, but should come in from the config as a eval capable string that
  // a utility method perhaps can handle per brand.
  if (config.colorDetailPageRoot) {
    return (
      <a href={`${config.colorDetailPageRoot}/${color.brandKey}${color.colorNumber}-${kebabCase(color.name)}`} {...other}>
        <FormattedMessage id='VIEW_DETAILS' />
      </a>
    )
  }

  if (config.displayDetailsLink && detailsLink) {
    return (
      <Link to={detailsLink} {...other}>
        <FormattedMessage id='VIEW_DETAILS' />
      </Link>
    )
  }

  return null
}

export default React.memo<Props>(DetailsLink)
