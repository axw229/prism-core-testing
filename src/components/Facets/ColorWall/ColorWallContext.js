// @flow
import React from 'react'
import { type GridBounds } from './ColorWall.flow'
import noop from 'lodash/noop'

export type ColorWallA11yContextProps = {
  a11yFocusCell: number[] | void,
  a11yFocusChunk: GridBounds,
  a11yFromKeyboard: boolean,
  a11yFocusOutline: boolean,
  a11yMaintainFocus: boolean
}

export const colorWallA11yContextDefault: ColorWallA11yContextProps = {
  a11yFocusCell: undefined,
  a11yFocusChunk: undefined,
  a11yFromKeyboard: false,
  a11yFocusOutline: false,
  a11yMaintainFocus: false
}

export type ColorWallContextProps = {
  addButtonText?: string,
  chunkClickable?: boolean,
  chunkMiniMap?: boolean,
  colorDetailPageRoot?: string,
  colorNumOnBottom?: boolean,
  colorWallBgColor: string,
  colorWallChunkPageRoot?: string,
  displayAddButton?: boolean,
  displayAddButtonText?: boolean,
  displayDetailsLink?: boolean,
  displayInfoButton?: boolean,
  hiddenSections?: string[],
  loading: boolean,
  swatchMaxSize: number,
  swatchSizeZoomed: number,
  swatchMinSize: number,
  swatchMinSizeZoomed: number,
  updateA11y: Function
}

export const colorWallContextDefault: ColorWallContextProps = {
  addButtonText: undefined,
  chunkClickable: false,
  chunkMiniMap: false,
  colorDetailPageRoot: undefined,
  colorNumOnBottom: false,
  colorWallChunkPageRoot: undefined,
  colorWallBgColor: '#EEEEEE',
  displayAddButton: false,
  displayAddButtonText: false,
  displayDetailsLink: true,
  viewColorText: false,
  displayInfoButton: false,
  hiddenSections: [],
  loading: false,
  swatchMaxSize: 33,
  swatchSizeZoomed: 50,
  swatchMinSize: 14,
  swatchMinSizeZoomed: 50,
  updateA11y: noop
}

const ColorWallContext = React.createContext<Object>({
  ...colorWallA11yContextDefault,
  ...colorWallContextDefault
})

export default ColorWallContext
