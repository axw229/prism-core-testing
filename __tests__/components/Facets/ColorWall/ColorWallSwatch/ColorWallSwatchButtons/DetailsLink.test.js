/* eslint-env jest */
import React from 'react'
import DetailsLink from 'src/components/Facets/ColorWall/ColorWallSwatch/ColorWallSwatchButtons/DetailsLink'
import { Link } from 'react-router-dom'
import { FormattedMessage } from 'react-intl'
import ColorWallContext from 'src/components/Facets/ColorWall/ColorWallContext'

const color = { id: 1, brandKey: 'sw', colorNumber: 1234, name: 'red' }

test('matches snapshot', () => {
  expect(mocked(<DetailsLink color={color} />)).toMatchSnapshot()
})

test('should render nothing when ColorWallContext.displayDetailsLink is false', () => {
  expect(mocked(
    <ColorWallContext.Provider value={{ displayDetailsLink: false }}>
      <DetailsLink color={color} />
    </ColorWallContext.Provider>
  ).html()).toBe("")
})

test('should render an anchor tag when ColorWallContext.colorDetailPageRoot has a value', () => {
  const expectedLink = `/homeowners/color/find-and-explore-colors/paint-colors-by-family/${color.brandKey}${color.colorNumber}-${color.name}`
  expect(mocked(
    <ColorWallContext.Provider
      value={{
        displayDetailsLink: true,
        colorDetailPageRoot: '/homeowners/color/find-and-explore-colors/paint-colors-by-family'
      }}
    >
      <DetailsLink color={color} />
    </ColorWallContext.Provider>
  ).find('a').prop('href')).toEqual(expectedLink)
})

test('should render Link to /active/color-detail/id/brandkey-colorNumber-name by default', () => {
  const expectedLink = `/active/color-detail/${color.id}/${color.brandKey}-${color.colorNumber}-${color.name}`
  expect(mocked(<DetailsLink color={color} />).find(Link).prop('to')).toEqual(expectedLink)
})

test('should render FormattedMessage component', () => {
  expect(mocked(<DetailsLink color={color} />).find(FormattedMessage).exists()).toBeTruthy()
})

test('should render FormattedMessage component with id defined as constant formattedMessageId', () => {
  const formattedMessageId = 'VIEW_DETAILS'
  expect(mocked(<DetailsLink color={color} />).find(FormattedMessage).prop('id')).toEqual(formattedMessageId)
})
