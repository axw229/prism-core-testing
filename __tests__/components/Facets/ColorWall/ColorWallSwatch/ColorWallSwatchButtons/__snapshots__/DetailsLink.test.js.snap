// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsLink should match snapshot 1`] = `
<Link
  to="/test"
>
  <FormattedMessage
    id="VIEW_DETAILS"
    values={Object {}}
  />
</Link>
`;

exports[`DetailsLink with config.ColorWall.displayViewDetails undefined should match snapshot 1`] = `""`;
